ttljob: paginate ranges when doing TTL

otan opened the pull request:
This commit batches ranges instead of scanning them in all at once. The
range batch size is controlled by the cluster setting
`sql.ttl.range_batch_size`. I don't anticipate this needing to be
controlled per table.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76726)
<!-- Reviewable:end -->

rafiss commented on pkg/sql/ttl/ttljob/ttljob.go:
```diff
@@ -56,6 +55,13 @@ var (
+	rangeBatchSize = settings.RegisterIntSetting(
+		settings.TenantWritable,
+		"sql.ttl.range_batch_size",
+		"amount of ranges to fetch at a time for a table",
```
maybe add "during deletion" somewhere in or at the end of the description

rafiss commented on pkg/sql/ttl/ttljob/ttljob_test.go:
```diff
@@ -317,7 +317,11 @@ func TestRowLevelTTLJobRandomEntries(t *testing.T) {
 			})
 			defer cleanupFunc()
 
+			rangeBatchSize := 1 + rng.Intn(3)
```
if the default is 100, should the test be closer to that?

otan replied:
we used a smaller number here here because we have less ranges (maximum of ~8).

rafiss commented on pkg/sql/ttl/ttljob/ttljob_test.go:
```diff
@@ -317,7 +317,11 @@ func TestRowLevelTTLJobRandomEntries(t *testing.T) {
 			defer cleanupFunc()
 
+			rangeBatchSize := 1 + rng.Intn(3)
+			t.Logf("range batch size: %d", rangeBatchSize)
```
random thought - we should have one place to randomize + log all the TTL settings

otan replied:
there is more or less this above -- it's just this is the only one randomised by a cluster setting

rafiss reviewed (approved):
lgtm! had some comments for future and a nit about the setting description

i'm not very familiar with this part of the KV api (like RangeMetaKey or MustAddr), so if needed, feel free to reach out to them for another review

otan commented:
bors r=rafiss

thanks!

otan commented:
bors r=rafiss

otan committed:
ttljob: paginate ranges when doing TTL

This commit batches ranges instead of scanning them in all at once. The
range batch size is controlled by the cluster setting
`sql.ttl.range_batch_size`. I don't anticipate this needing to be
controlled per table.

Release note: None

otan commented:
bors r=rafiss
